axelrod/tests/strategies/test_titfortat.py

```diff
@@ -717,3 +717,50 @@ def test_varying_memory_depth(self):
         self.assertEqual(self.player(1, 1).classifier['memory_depth'], 1)
         self.assertEqual(self.player(0, 3).classifier['memory_depth'], 3)
         self.assertEqual(self.player(5, 3).classifier['memory_depth'], 5)
+
+
+class TestMichaelos(TestPlayer):
+    """
+    Tests for the Michaelos strategy
+    """
+
+    name = "Michaelos: (D,)"
+    player = axelrod.Michaelos
+    expected_classifier = {
+        'memory_depth': 1,
+        'stochastic': True,
+        'makes_use_of': {'length'},
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+
+    def test_strategy(self):
+
+        actions = [(C, C), (C, C), (C, C), (D, C)]
+        self.versus_test(axelrod.Cooperator(), expected_actions=actions,
+                         attrs={"is_defector": False}, seed=2)
+
+        actions = [(C, C), (C, C), (C, C), (C, C)]
+        self.versus_test(axelrod.Cooperator(), expected_actions=actions,
+                         attrs={"is_defector": False},
+                         match_attributes={"length": -1}, seed=2)
+
+        actions = [(C, D), (D, D), (D, D), (D, D)]
+        self.versus_test(axelrod.Defector(), expected_actions=actions,
+                         attrs={"is_defector": False}, seed=2)
+
+        actions = [(C, D), (D, D), (D, D), (D, D)]
+        self.versus_test(axelrod.Defector(), expected_actions=actions,
+                         attrs={"is_defector": False},
+                         match_attributes={"length": -1}, seed=2)
+
+        # Chance of becoming a defector is 50% after (D, C) occurs.
```

Comment:
> After (C, D) Michaelos will play D since it is still playing like Tit-for-Tat.

Sorry you're correct, that was a typo from me, I simply mean to include a random seed version to illustrate that 50% of the time after `(D, C)` (not `(C, D)` - sorry) the strategy will cooperate **and ** `is_defector` will be False (so kind of like your `seed=1` test but without the second `(C, D)`.

> For the second test, does tweaking the test mean changing the expected actions?

Yes.